src/state/slices/stakingDataSlice/stakingDataSlice.ts

```diff
@@ -103,6 +148,68 @@ export const stakingDataApi = createApi({
           dispatch(stakingData.actions.setStatus('loaded'))
         }
       }
+    }),
+    getAllValidatorsData: build.query<Validators, AllValidatorDataArgs>({
+      queryFn: async ({ chain }, { dispatch }) => {
+        const chainAdapters = getChainAdapters()
+        const adapter = (await chainAdapters.byChain(
+          chain
+        )) as CosmosSdkBaseAdapter<ChainTypes.Cosmos>
```

Comment:
https://github.com/shapeshift/web/blob/develop/src/state/slices/txHistorySlice/txHistorySlice.ts#L304-L313

safer more modern way to do this with chainId